app/services/puppet_fact_parser.rb

```diff
@@ -5,9 +5,14 @@ def operatingsystem
     orel = os_release.dup
 
     if orel.present?
-      major, minor = orel.split('.', 2)
-      major = major.to_s.gsub(/\D/, '')
-      minor = minor.to_s.gsub(/[^\d\.]/, '')
+      if os_name =~ /ubuntu/i
+        major = os_major_version
+        minor = os_minor_version
```

Comment:
I've been debating making this the behavior everywhere and would be OK with trying that.

Reply:
So that we simply leave out the if / else part and use `os_major_version` and `os_minor_version` for all operating systems? I also noticed that the `if orel.present?` isn't necessary here anymore, as the whole part from line 8 to line 24 is no longer dependent on it.

Reply:
indeed, get rid of orel altogether.

Reply:
I had a look at relying on the `major` and `minor` versions reported by facter and it doesn't fit for all operating systems. The `os_release` often uses the `full` release reported by facter and parses it as necessary. Relying on the reported `major` and `minor` versions breaks some of the OSes (like Solaris and CentOS 8).

IMO it would be best to rework detection of the OS version in the PuppetFactParser in a separate PR.